include/alpaka/core/Common.hpp

```diff
@@ -102,3 +102,17 @@
 #else
 #    define ALPAKA_STATIC_ACC_MEM_CONSTANT
 #endif
+
+//! This macro disables memory optimizations for annotated device memory.
+//!
+//! Example:
+//!   ALPAKA_DEVICE_VOLATILE float* ptr;
+//!
+//! This is useful for pointers, (shared) variables and shared memory which are used in combination with
+//! the alpaka::mem_fence() function. It ensures that memory annotated with this macro will always be written directly
+//! to memory (and not to a register or cache because of compiler optimizations).
+#if(BOOST_LANG_CUDA || BOOST_LANG_HIP)
+#    define ALPAKA_DEVICE_VOLATILE volatile
```

Comment:
If you only like to activate on the GPU device but not on cpu you must write 

```
#if(BOOST_LANG_CUDA && BOOST_ARCH_PTX) || (BOOST_LANG_HIP && __HIP_DEVICE_COMPILE__=
```

Reply:
Done.